### ISSUE TITLE
Firefox - Cookie unit tests fail for asserts on non-standard properties (eg sameParty) 

### ISSUE THREAD

**Issue by juliandescottes** [puppeteer/puppeteer#6976]
Puppeteer recently started to assert new cookie properties in unit tests:
- `sourcePort` & `sourceScheme` in https://github.com/puppeteer/puppeteer/pull/6927
- `sameParty` in https://github.com/puppeteer/puppeteer/pull/6797

They seem non-standard. At least for `sameParty` I could find a reference clearly stating that Firefox would not support it: https://www.chromestatus.com/feature/5280634094223360 
For sourcePort & sourceScheme I didn't find any resource.

Firefox currently has no way of supporting such properties, so all the tests asserting those values will fail on Firefox when we update Puppeteer:
- https://github.com/puppeteer/puppeteer/blob/35e66c43040a41d7e47578044688f368f366c5bb/test/cookies.spec.ts#L34
- https://github.com/puppeteer/puppeteer/blob/35e66c43040a41d7e47578044688f368f366c5bb/test/cookies.spec.ts#L90
- https://github.com/puppeteer/puppeteer/blob/35e66c43040a41d7e47578044688f368f366c5bb/test/defaultbrowsercontext.spec.ts#L27
(only mentioning the tests which used to pass before asserting the new properties)

I realize that in puppeteer those tests were already running as `itFailsFirefox` before, so the regression was not detected. But they were actually passing and we are running them successfully in the Firefox CI today

Firefox will probably never be able to pass the assert for those properties, but it would be great to be able to still run the tests, maybe we could check the non-standard properties only when running in Chrome?

(corresponding Bugzilla reference: https://bugzilla.mozilla.org/show_bug.cgi?id=1694506)

**Comment by whimboo** [comment]
@jschfflr and @mathiasbynens could we get those extra checks put behind an if condition for Chrome?

**Comment by juliandescottes** [comment]
FWIW, I tried a simple modification to skip `sourcePort`, `sourceScheme` and `sameParty` from the asserts on cookies when `isChrome` is false and this fixes the tests for Firefox, so it's really just about those three properties ([changeset](https://hg.mozilla.org/try/rev/e8da1a77883362c631bc2c3ab622502290d62dbd), [ci job](https://treeherder.mozilla.org/jobs?repo=try&revision=427da6f8fe1cf60e21598dab9026341c54fc040e)).

Mostly extracted the logic to a small helper to assert cookies.
```javascript
export const expectCookieEquals = (cookies, expectedCookies) => {
  const { isChrome } = getTestState();
  if (!isChrome) {
    for (const expectedCookie of expectedCookies) {
      // Remove non-standard properties when testing another browser than Chrome.
      delete expectedCookie.sameParty;
      delete expectedCookie.sourcePort;
      delete expectedCookie.sourceScheme;
    }
  }

  expect(cookies).toEqual(expectedCookies);
};
```

**Comment by jschfflr** [comment]
Sounds good to me! Feel free to upload your change as a pull request here and I'm happy to accept it!

**Comment by juliandescottes** [comment]
Thanks! opened https://github.com/puppeteer/puppeteer/pull/6994

**Comment by juliandescottes** [comment]
This can be closed as #6994 was merged
